refactor(wasm): Refactor wasm component adapter and wasi/wit code

stevenj opened the pull request:
Refactor to move wasi outside the wasi component adapter crate.
Rename the component adapter crate so it can not be confused with the official one.
Remove the ability to build a command version as hermes does not support it.

Mr-Leshiy reviewed (commented):
Why to keep `wasi/wit` files in two places ?
Maybe it make sense to get rid of one copy of them ?

stevenj commented on Earthfile:
```diff
@@ -18,7 +18,18 @@ markdown-check-fix:
- 
+
+# check-spelling Check spelling in this repo inside a container.
+spell-list-words:
```
This is a utility test target which just lists out all words which fail spelling for a particular path.
It's how I created a custom dictionary that only effects the wasm code.
I left it in because its useful, and is not used unless specifically called but a user.

stevenj commented:
> Why to keep `wasi/wit` files in two places ? Maybe it make sense to get rid of one copy of them ?

We don't.  I fixed the import path to use a relative reference and fixed the COPY in earthly to put the files at the right point in the build.

stevenj commented on wasm/.cspell.json:
```diff
@@ -0,0 +1,283 @@
+{
```
Yes, thats the correct name.

Mr-Leshiy reviewed (approved):
LGTM
